doc/kernel/usermode/usermode.rst

```diff
@@ -9,6 +9,144 @@ are defined, and how memory may be managed to support user mode threads.
 For details on creating threads that run in user mode, please see
 :ref:`lifecycle_v2`.
 
+Threat Model
+============
+
+For threads running in a non-privileged CPU state (hereafter referred to as
+'user mode') we aim to protect against the following:
+
```

Comment:
It still feels like we are delving into a discussion of what was implemented rather than threat model and goals.  Maybe moving up the use cases first would help, and/or adding some high level text, ala "User mode threads are considered to be untrusted by Zephyr and are therefore isolated from other user mode threads and from the kernel.  A flawed or malicious user mode thread cannot leak or modify the private data/resources of another user mode thread or the kernel and cannot interfere with or control another user mode thread or the kernel."